### PR TITLE
Add internal/config/index test

### PR DESCRIPTION
**Pull request by hlts2**
### Description:

I added test for `internal/config/index.go` and godoc comment.

**grammer check passed**

### Environment:

- Go Version: 1.16
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/config/index.go [hunk 1]
```diff
@@ -17,6 +17,7 @@
 // Package config providers configuration type and load configuration logic
 package config
 
+// Indexer represents Indexer configuration.
```
Comment (hlts2) [inline internal/config/index.go:20]:
```suggestion
// Indexer represents the Indexer configurations.
```

File: internal/config/index_test.go [hunk 2]
```diff
@@ -56,60 +59,142 @@ func TestIndexer_Bind(t *testing.T) {
...
+						AgentDNS:               "vald-agent-ngt.vald.svc.cluster.local",
+						Concurrency:            10,
+						AutoIndexDurationLimit: "30m",
+						AutoIndexCheckDuration: "1m",
+						AutoIndexLength:        100,
+						CreationPoolSize:       10000,
+						NodeName:               "vald-01-worker",
+						Discoverer: &DiscovererClient{
+							Client:             newGRPCClientConfig(),
+							AgentClientOptions: newGRPCClientConfig(),
```
Comment (vankichi) [inline internal/config/index_test.go:123]: same as #1122

**Review by vankichi — APPROVED** [review]
LGTM

**Review by kevindiu — APPROVED** [review]
LGTM

**Pull request merged**